engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.content-hosts.controller:ContentHostsBulkSystemPurposeModalController
+ *
+ * @requires $scope
+ * @requires $uibModalInstance
+ * @requires HostBulkAction
+ * @requires Organization
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires hostIds
+ *
+ * @description
+ *   A controller for providing bulk action functionality for setting system purpose values
+ */
+angular.module('Bastion.content-hosts').controller('ContentHostsBulkSystemPurposeModalController',
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            return org.system_purposes.addons.length;
+        });
+
+        $scope.purposeAddonsList = function () {
+            if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
+                $scope.defaultOptions = ['No Change', 'None (Clear)'];
+                $scope.addons = $scope.organization.system_purposes.addons.map(function (addon) {
+                    return addon; });
```

Comment:
If `$scope.defaultOptions` isn't used anywhere outside this function, it can just be a regular variable.  Same with `$scope.addons`:
```js
var defaultOptions = ['No Change', 'None (Clear)'];
var addons;
if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
                ...
                $scope.addOnList = defaultOptions.concat(addons);
```

actually, `addons` and `addOnList` probably aren't needed at all:  
* Doing this `.map` on `addons` would just return `$scope.organization.system_purposes.addons` unaltered, so there's no reason to do the map
* `$scope.addonList` is assigned just before returning it, so instead of `return $scope.addonList` we can just return the expression directly without assigning it to a scope property.

```js
return $scope.defaultOptions.concat($scope.organization.system_purposes.addons);

Reply:
Much cleaner now :)